pyxem/generators/indexation_generator.py

```diff
@@ -91,33 +102,40 @@ def correlate(self,
         signal = self.signal
         library = self.library
 
+        method_dict = { 'fast_correlation' : fast_correlation,
+                     'normalized_correlation' : normalized_correlation
+                     }
+
         if mask is None:
             # Index at all real space pixels
             mask = 1
 
-        # TODO: Add extra methods
-        no_extra_methods_yet = True
-        if no_extra_methods_yet:
-            # adds a normalisation to library
-            for phase in library.keys():
-                norm_array = np.ones(library[phase]['intensities'].shape[0])  # will store the norms
-                for i, intensity_array in enumerate(library[phase]['intensities']):
-                    norm_array[i] = np.linalg.norm(intensity_array)
-                library[phase]['pattern_norms'] = norm_array  # puts this normalisation into the library
-
-            matches = signal.map(correlate_library,
-                                 library=library,
-                                 n_largest=n_largest,
-                                 mask=mask,
-                                 inplace=False,
-                                 **kwargs)
+        chosen_function = select_method_from_method_dict(method,method_dict,print_help)
+        #Not used, but the function tests if selected method is a valid argument, and can print help for selected method.
```

Comment:
Comments should be above the line they refer to, also I would cut all the words before "test"